src/Neo/SmartContract/Manifest/ContractPermissionDescriptor.cs

```diff
@@ -114,7 +114,8 @@ public bool Equals(ContractPermissionDescriptor other)
             if (this == other) return true;
             if (IsWildcard == other.IsWildcard) return true;
             if (IsHash) return Hash.Equals(other.Hash);
-            else return Group.Equals(other.Group);
+            if (IsGroup) return Group.Equals(other.Group);
```

Comment:
The logic changed, not only was added UT

Reply:
Because in the original code, the `Group` may be null, and there will be an exception

Reply:
This behaviour change should be done in a different PR

Reply:
fixed

Reply:
https://github.com/neo-project/neo/pull/3396